src/common/etcd_delegate.rs

```diff
@@ -111,12 +116,96 @@ impl EtcdDelegate {
         })?;
 
         let kvs = resp.take_kvs();
+
         match kvs.get(0) {
-            Some(kv) => Ok(Some(util::decode_from_bytes::<T>(kv.value())?)),
+            Some(kv) => Ok(Some((
+                util::decode_from_bytes::<T>(kv.value())?,
+                kv.version(),
+            ))),
             None => Ok(None),
         }
     }
 
+    /// offer an version check
+    async fn write_to_etcd_with_version<
+        T: DeserializeOwned + Serialize + Clone + Debug + Send + Sync,
+        K: Into<Vec<u8>> + Debug + Clone + Send,
+    >(
+        &self,
+        key: K,
+        value: &T,
+        prev_version: KVVersion,
+        timeout: Option<Duration>,
+    ) -> DatenLordResult<Option<(T, KVVersion)>> {
+        let bin_value = bincode::serialize(value)
+            .with_context(|| format!("failed to encode {value:?} to binary"))?;
+        let mut put_request = etcd_client::EtcdPutRequest::new(key.clone(), bin_value);
+        if let Some(dur) = timeout {
+            put_request.set_lease(
+                self.etcd_rs_client
+                    .lease()
```

Comment:
Why does the 'put' operation here require a lease() to ensure?

Reply:
Lease is for timeout.

Reply:
So why does the `put` operation here require a timeout?

Reply:
That's common, sometimes a value is just for implement lock, avoid conflict operation by txn or temporary use, we don't want to keep it in kv all the time.